snuba/replacer.py

```diff
@@ -454,12 +460,13 @@ def _build_topic_group_index_key(
     ) -> str:
         """
         Builds a unique key for a message being processed for a specific
-        topic, consumer group, and partition.
+        consumer group and partition.
         """
         return ":".join(
             [
                 "replacement",
-                message_metadata.topic_name,
+                self.__consumer_group,
+                self.__replacements_topic,
```

Comment:
You don't need a replacements topic in the key. The reason being there wouldn't be multiple replacement topics. All the consumer groups would work on the same topic. 

You can remove the storing/passing of topics from the PR altogether.